crates/aptos-logger/src/aptos_logger.rs

```diff
@@ -494,7 +495,7 @@ impl LoggerService {
                             .enabled(&entry.metadata)
                         {
                             let s = (self.facade.formatter)(&entry).expect("Unable to format");
-                            printer.write(s)
+                            printer.write_buferred(&mut buffer, s);
```

Comment:
where do you flush it?

Reply:
The [BufWriter](https://doc.rust-lang.org/std/io/struct.BufWriter.html) keeps an in-memory buffer of data and writes it to an underlying writer in large, infrequent batches.

BufWriter<W> can improve the speed of programs that make small and repeated write calls -- that's our use-case.
An additional flush will happen when the writer is dropped, when the program dies.

Reply:
the api doesn't look right to me, the buffer should be part of the `StdoutWriter` instead of passing a &mut in

Reply:
Seems like the buffer size is 8 KB by default -  do we need to tune that as well?

Reply:
Making Bufwriter a part if StdoutWriter which is a Filed inside the LOGGER will necessitate passing a `&mut` of the logger and thus necessitate adding a Mutex around the Arc. This Mutex would be a single contention point as all  threads would compete to grab it. Initially this may not be an issue as all threads are already serialized on the Aptos-Channel.

Reply:
IMHO dumping the loggs once in 8K is good enough. I wouldn't change it unless there is a good reason.

Reply:
agree with @zekun000 , printer should be hiding this.

cool stuff though, does it impact perf at all?

Reply:
StdoutWriter is a Box instead of an Arc afaict, I don't see a problem of using &mut ref of it?

Reply:
Re Perf. It does free up some cpu cycles, it seems to add 200~300TPS. So net benefit is between 0%~5% on a good day.